enhancements/operator-hub-check.md

```diff
@@ -0,0 +1,181 @@
+
+#### New Criteria valid which will be checked against to a specific Kubernetes version
+
+For the checks which should be done accordingly for some specific version(s) it will use the Kubernetes version which has been provided to the validator via flag or via the bundle operator configuration. For example, if we run:
+
+```sh
+ operator-sdk bundle validate ./bundle \
+     --select-optional name=operatorhub \
+     --optional-values="k8s=1.22"
+```
+
+Then, it means that the check will validate the operator bundle configuration accordingly to the Kubernetes version informed where it is intend to be published to. In this case, the validator will consider the Kubernetes version `=1.22`. 
+
+If the `--optional-values="k8s=1.22"` not be used then, we will use the `minKubeVersion` in the CSV to do the checks. So, the criteria is `>=minKubeVersion`. 
+
+By last, if the version is not provided then we should consider as described above the operator bundle is intend to work well in any Kubernetes version.
+
+**CRD and Webook APIS deprecated and unsupported versions**
+
+Note that in these cases, users should be informed that the `apiextensions/v1beta1` and that the `admissionregistration.k8s.io/v1beta1` was deprecated in Kubernetes `1.16` and will/was removed in `1.22`.
+
+- To check if the operator bundle is using a CRD API version which is deprecated or not supported we will need to see if the CRD manifests in the bundle are using `apiVersion: apiextensions.k8s.io/v1beta1`. (See [here](https://github.com/operator-framework/operator-sdk/blob/v1.4.2/testdata/go/v2/memcached-operator/bundle/manifests/cache.example.com_memcacheds.yaml#L1)). Note that we need to check all CRD(s) added on the bundle.
+
+- To check if the operator bundle is using the Webhook API version which is deprecated we will need to see if the CSV has the spec `webhookdefinitions.admissionReviewVersions.v1beta1`. (See [here](https://github.com/operator-framework/operator-sdk/blob/v1.4.2/testdata/go/v2/memcached-operator/bundle/manifests/memcached-operator.clusterserviceversion.yaml#L203-L205)
+
+The error raised will be accordingly with the Kubernetes version that we have, see:
+
+| Conditional | Type |
+| ------ | ----- |
+| operator bundle supports k8s => `1.22` | error |
+| operator bundle supports k8s `=> 1.16 <= 1.22` | warning |
+
+**IMPORTANT** 
+
+> Until Kubernetes `1.22` not be released and OpenShift has not a release using that we should return `warning` and NOT `error` in any scenario where the `--optional-values="k8s=1.22"` flag/value is not provided. 
+
+Then, it means that:
+- `--optional-values="kubeVersion=value"` flag with a value `=> 1.16 <= 1.22` we will raise warning.
+- `--optional-values="kubeVersion=value"` flag with a value `=> 1.22` we will raise error.
+- Any `minKubeVersion` or `empty` means warning 
```

Comment:
why wouldn't it be an error if the minKubeVersion is 1.22+?  Sure, it's silly to ship an operator with minkubeversion greater than the current latest kubeversion, but maybe they are planning ahead, or experimenting.

Reply:
Because the apis are no longer supported after 1.22.
1.22 was not released yet. So, we cannot raise an error here. 

Before the release of 1.22:
- Any `minKubeVersion` or `empty` means to raise a warning 
After the release of 1.22
Any `minKubeVersion` means to raise an error.  

I will clarify.